ui/src/app/layout/topics/schema-section.component.ts

```diff
@@ -48,13 +48,25 @@ export class SchemaSectionComponent implements OnInit {
     }
 
     ngOnInit(): void {
-        combineLatest([this.topic, this.environmentsService.getCurrentEnvironment()]).subscribe({
-            next: value => {
-                if (value[0]) {
-                    this.loadSchemas(value[0], value[1].id);
+        this.environmentsService.getCurrentEnvironment().subscribe({
+            next: env => {
+                if (this.topic && env) {
+                    this.loadSchemas(this.topic, env.id);
                 }
             }
         });
+
+        this.selectedEnvironment = this.environmentsService.getCurrentEnvironment();
+    }
+
+    async ngOnChanges(changes: SimpleChanges) {
+        if (changes.topic) {
+            const change = changes.topic;
+            if (change.currentValue) {
+                const env = await this.environmentsService.getCurrentEnvironment().toPromise();
```

Comment:
Hmm, this should read
`getCurrentEnvironment().pipe(take(1)).toPromise()` - otherwise, that Promise would never resolve.